docs/book/features/volume_snapshot.md

```diff
@@ -0,0 +1,208 @@
+<!-- markdownlint-disable MD033 -->
+<!-- markdownlint-disable MD034 -->
+# vSphere CSI Driver - Volume Snapshot & Restore
+
+- [Introduction](#introduction)
+- [Prerequisite](#prereq)
+- [How to enable Volume Snapshot & Restore feature in vSphere CSI](#how-to-deploy)
+- [How to use Volume Snapshot & Restore feature](#how-to-use)
+- [Configuration - Maximum Number of Snapshots per Volume](#config-param)
+
+## Introduction <a id="introduction"></a>
+
+CSI Volume Snapshot & Restore feature was introduced as an alpha feature in Kubernetes 1.12, promoted to beta in Kubernetes 1.17 and moved to GA in Kubernetes 1.20.
+Volume Snapshot & Restore feature will be added in vSphere CSI driver 2.4 as an Alpha feature.
+
+Known limitations for the Alpha feature in vSphere CSI driver 2.4 are listed below.
+
+1. It is only supported in ReadWriteOnce volumes based on First Class Disk, i.e., FCD or CNS block volume, while not yet supported in ReadWriteMany volumes based on vSAN file service.
+2. It is only supported in [Vanilla Kubernetes](https://github.com/kubernetes/kubernetes) cluster now, while not yet supported in either [vSphere with Kubernetes](https://blogs.vmware.com/vsphere/2019/08/introducing-project-pacific.html) cluster aka Supervisor Cluster or [Tanzu Kubernetes Grid Service](https://blogs.vmware.com/vsphere/2020/03/vsphere-7-tanzu-kubernetes-clusters.html) cluster aka Guest Cluster.
+3. Volume restore can only create a PVC with the same storage capacity as the source VolumeSnapshot.
+4. vSphere CSI introduces a constraint on the maximum number of snapshots per ReadWriteOnce volume. The maximum is configurable but set to 3 by default. Please refer to the section of [Configuration - Maximum Number of Snapshots per Volume](#config-param) for more detail.
```

Comment:
What about cannot delete volume and expand volume where there are snapshots.  Those are also limitations.